aggregator/entry.go

```diff
@@ -296,10 +319,11 @@ func (e *Entry) TryExpire(now time.Time) bool {
 		return false
 	}
 	e.closed = true
-	for p, agg := range e.aggregations {
-		agg.Value.(metricElem).MarkAsTombstoned()
-		delete(e.aggregations, p)
+	for i := range e.aggregations {
+		e.aggregations[i].elem.Value.(metricElem).MarkAsTombstoned()
+		e.aggregations[i] = aggregationValue{}
```

Comment:
How come we're resetting the the entries here when we truncate the slice at the end of the loop? Is this to release any internal pointers?

Reply:
Correct, even if you reset the slice, the backing array still holds pointers to internal fields, so this is commonly done to avoid GC sweeping overhead.

Reply:
Sure will do.